src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1333,36 +1336,38 @@ internal static bool IsCustomTaskType(this NamedTypeSymbol type, out NamedTypeSy
         /// </summary>
         internal static TypeSymbol NormalizeTaskTypes(this TypeSymbol type, CSharpCompilation compilation)
         {
-            NormalizeTaskTypesCore(compilation, ref type);
+            NormalizeTaskTypesInType(compilation, ref type);
             return type;
         }
 
         /// <summary>
         /// Replace Task-like types with Task types. Returns true if there were changes.
         /// </summary>
-        private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref TypeSymbol type)
+        private static bool NormalizeTaskTypesInType(CSharpCompilation compilation, ref TypeSymbol type)
         {
             switch (type.Kind)
```

Comment:
Ah didn't realize ErrorTypes could nest types in that fashion.  I'd thought of them as non-generic flat types.  If not then yeah we should dig into them too if that's possible.  

Same for pointers.  Made me wonder if we have tests for pointers referring to task like types though.  @cston? 

---
In reply to: [74292992](https://github.com/dotnet/roslyn/pull/12671#discussion_r74292992) [](ancestors = 74292992,74289912,74289506,74288645,74285338)